### ISSUE TITLE
New Lint: parameter of function only used in recursive call

### ISSUE THREAD

**Issue by flip1995** [rust-lang/rust-clippy#2624]
If a parameter of a function is only used in a recursive call of this functions body, it is probably unintended and useless:
```rust
fn rec_fn(a: i32) {
    //some conditions, no a... 
    rec_fn(a) 
} 
```
See also https://github.com/rust-lang-nursery/rust-clippy/pull/2622#comment-178952187

**Comment by jamesmcm** [comment]
This should be addressed by #8422

**Comment by xFrednet** [comment]
Thank you for pointing that out @jamesmcm
